### DISCUSSION TITLE
Unable to get button click event example to work

### DISCUSSION THREAD

**Discussion by Tris-an** [AvaloniaUI/Avalonia#12359] [Q&A] [answered]
I have recently started using Avalonia UI and am relatively new to anything xaml. 
I am using the visual studio extension and v11. However I am unable to get the most basic button event to work.
All I have done is create a blank .net C# project (not MVVM) and then add the example code from the button doc page to the main window.  The moment I add Click="ClickHandler" I receive the error below. If I create a new project using v0.10 the identical code does work.
Any help would be appreciated.

Unable to find suitable setter or adder for property Click of type Avalonia.Controls:Avalonia.Controls.Button for argument System.Runtime:System.String, available setter parameter lists are:
System.EventHandler`1<Avalonia.Interactivity.RoutedEventArgs> Line 10, position 11.

https://docs.avaloniaui.net/docs/next/reference/controls/buttons/button

**Comment by IanRawley** [comment, marked as answer]
That usually means that you don't have a public method with the correct signature in the code behind.  
Make sure your view codebehind has:
```
public void ClickHandler(object? sender, RoutedEventArgs e) {  }
```

**Reply by IanRawley** [reply]
For reference this part 
```
System.EventHandler`1<Avalonia.Interactivity.RoutedEventArgs> Line 10, position 11.
```
tells you what the second argument for your handler should be if you're ever unsure.

[a comment by an automated account is not shown]

**Reply by thevortexcloud** [reply]
Visual Studio lies/gets confused a lot. Just don't trust it completely and always check the actual build log, and you will be fine.
